app/web/features/messages/groupchats/GroupChatHeaderBar.tsx

```diff
@@ -0,0 +1,285 @@
+import { groupChatsRoute } from "routes";
+import { service } from "service";
+import { theme } from "theme";
+
+const StyledTitleBox = styled("div")({
+  flexGrow: 1,
+  width: "100%",
+  display: "flex",
+  alignItems: "center",
+  marginInlineEnd: theme.spacing(2),
+  marginInlineStart: theme.spacing(2),
+  "& > *": { marginInlineEnd: theme.spacing(2) },
+});
+
+export default function GroupChatHeaderBar({
+  chatId,
+  currentUserId,
+  groupChat,
+  groupChatMembersQuery,
+  title,
+}: {
+  chatId: number;
+  currentUserId: number;
+  groupChat: GroupChat.AsObject | undefined;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  groupChatMembersQuery: any;
+  title: string | undefined;
+}) {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+  const { t } = useTranslation([GLOBAL, MESSAGES]);
+
+  const isChatAdmin = groupChat?.adminUserIdsList.includes(currentUserId);
+  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
+
+  const menuAnchor = useRef<HTMLButtonElement>(null);
+  const [isOpen, setIsOpen] = useState({
+    admins: false,
+    invite: false,
+    leave: false,
+    members: false,
+    menu: false,
+    mute: false,
+    settings: false,
+  });
+
+  const unmuteMutation = useMutation<void, RpcError>(
+    async () => {
+      await service.conversations.muteChat({
+        groupChatId: chatId,
+        unmute: true,
+      });
+    },
+    {
+      onSuccess() {
+        queryClient.setQueryData(groupChatKey(chatId), {
+          ...groupChat,
+          muteInfo: { muted: false },
+        });
+        queryClient.invalidateQueries(groupChatKey(chatId));
+      },
+    }
+  );
+
+  const handleBack = () => router.push(groupChatsRoute);
+
+  const handleClick = (item: keyof typeof isOpen) => {
+    //just unmute straight away with no dialog if muted
+    if (item === "mute" && groupChat?.muteInfo?.muted) {
+      unmuteMutation.mutate();
+      setIsOpen((prev) => ({ ...prev, menu: false }));
+      return;
+    }
+
+    //close the menu if a menu item was selected
+    if (item !== "menu") {
+      setIsOpen((prev) => ({ ...prev, [item]: true, menu: false }));
+    } else {
+      setIsOpen((prev) => ({ ...prev, [item]: true }));
+    }
+  };
+
+  const handleClose = (item: keyof typeof isOpen) => {
+    setIsOpen({ ...isOpen, [item]: false });
+  };
+
+  return (
+    <>
+      <HeaderButton
+        onClick={handleBack}
+        aria-label={t("messages:chat_view.back_button.a11y_label")}
+        size={isMobile ? "small" : "medium"}
+      >
+        <BackIcon fontSize={isMobile ? "small" : "medium"} />
+      </HeaderButton>
+
+      {groupChat?.isDm ? (
+        <StyledTitleBox>
+          <Link
+            href={`/user/${getDmUsername(
+              groupChatMembersQuery,
+              currentUserId
+            )}`}
+          >
+            <a>
+              <PageTitle
+                sx={{
+                  [theme.breakpoints.down("sm")]: {
```

Comment:
Use `isMobile` here?

Reply:
It's user for both, just I'm making the fontSize smaller than default for mobile. The CSS query needs the `theme.breakpoints.down("sm")` as it break down to `@media (max-width:599.95px)` for example when the function is run, versus a boolean in `useMediaQuery` hook.

Reply:
OK, makes sense